Build fails on Arch Linux

Fails during linking:
```
[ 31%] Linking CXX executable ../bin/alternating_tests
/usr/bin/ld: CMakeFiles/alternating_tests.dir/server.cpp.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/lib/libpthread.so.0: error adding symbols: DSO missing from command line
```
Fixed it by adding pthread to target_link_libraries in test/CMakeLists.txt but of course it's not very portable.

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,10 +1,12 @@
 macro(add_cpr_test _TEST_NAME)
+    find_package(Threads)
     add_executable(${_TEST_NAME}_tests
         ${_TEST_NAME}_tests.cpp server.cpp server.h)
     target_link_libraries(${_TEST_NAME}_tests
         ${GTEST_LIBRARIES}
         ${CPR_LIBRARIES}
-        ${MONGOOSE_LIBRARIES})
+        ${MONGOOSE_LIBRARIES}
+        ${CMAKE_THREAD_LIBS_INIT})
     add_test(NAME cpr_${_TEST_NAME}_tests COMMAND ${_TEST_NAME}_tests)
     # Group under the "tests" project folder in IDEs such as Visual Studio.
     set_property(TARGET ${_TEST_NAME}_tests PROPERTY FOLDER "tests")

